### PR TITLE
Verify requestid in StartWorkflowExecutionRequest

### PR DESCRIPTION
**Pull request by lihannan99**

### CODE REVIEW

File: service/history/historyEngine.go [hunk 1]
```diff
@@ -3273,6 +3273,9 @@ func validateStartChildExecutionAttributes(parentInfo *persistence.WorkflowExecu
 }
 
 func validateStartWorkflowExecutionRequest(request *workflow.StartWorkflowExecutionRequest, maxIDLengthLimit int) error {
+	if request.RequestId == nil || len(common.StringDefault(request.RequestId)) == 0 {
```
Comment (longquanzheng) [inline service/history/historyEngine.go]: I think checking len(request.requestID) is sufficient.

Comment (lihannan99) [reply service/history/historyEngine.go]: updated. Thanks!

**Comment by lihannan99** [comment]
This is fix to https://github.com/uber/cadence/issues/1487

File: service/history/historyEngine.go [hunk 2]
```diff
@@ -3274,6 +3274,9 @@ func validateStartChildExecutionAttributes(parentInfo *persistence.WorkflowExecu
 }
 
 func validateStartWorkflowExecutionRequest(request *workflow.StartWorkflowExecutionRequest, maxIDLengthLimit int) error {
+	if len(common.StringDefault(request.RequestId)) == 0 {
```
Comment (yycptt) [inline service/history/historyEngine.go]: Can we use len(request.GetRequestId()) == 0 here?

**Pull request merged**